athenzdomains.go

```diff
@@ -0,0 +1,183 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/yahoo/athenz/clients/go/zms"
+	v1 "github.com/yahoo/k8s-athenz-istio-auth/pkg/apis/athenz/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	replacer = strings.NewReplacer(".*", ".*", "*", ".*")
+)
+
+// CRMap - cr cache
+type CRMap struct {
+	RoleToPrincipals map[string][]*zms.RoleMember
+	RoleToAssertion  map[string][]SimpleAssertion
+}
+
+// SimpleAssertion - processed policy
+type SimpleAssertion struct {
+	resource *regexp.Regexp
+	action   *regexp.Regexp
+	effect   string
+}
+
+// Cache - cache for athenzdomains CR
+type Cache struct {
+	CrIndexInformer cache.SharedIndexInformer
+	DomainMap       map[string]CRMap
+	lock            sync.RWMutex
+}
+
+// BuildCache - generate new athenzdomains cr cache
+func BuildCache(crIndexInformer cache.SharedIndexInformer) *Cache {
```

Comment:
Could we instead of returning a cache object here initialize a private object in this package so we can avoid passing it around everywhere as all logic is now encapsulated here.